### ISSUE TITLE
"caffe.LayerParameter" has no field name "bn_param"

### ISSUE THREAD

**Issue by mercileesb** [yjxiong/temporal-segment-networks#139]
Hi,

I got a problem when I tested.

[libprotobuf ERROR google/protobuf/text_format.cc:245] Error parsing text-format caffe.NetParameter: 19:12: Message type "caffe.LayerParameter" has no field named "bn_param".
F1006 14:45:42.856966 15514 upgrade_proto.cpp:88] Check failed: ReadProtoFromTextFile(param_file, param) Failed to parse NetParameter file: models/ucf101/tsn_bn_inception_rgb_deploy.prototxt
*** Check failure stack trace: ***
Aborted (core dumped)

I fixed **bn_param** in caffe.prototxt to **batch_norm_param**.

but, I got a similar problem like ~~~has no field named "frozen"

Could you please help me?

**Comment by yjxiong** [comment]
` Message type "caffe.LayerParameter" has no field named "bn_param".`

This indicates you are running with another installation of Caffe. Please check your environment settings.

Also, the `BN` layer in our Caffe is different from `BatchNorm` layer in the official Caffe. We use the full implementation of batch normalization with the affine transformation attached, while `BatchNorm` only has the normalization part.

**Comment by Aravinda89** [comment]
Hi,

Got this error. Correct action caffe version is built in lib/caffe-action

frame folder analysis done
WARNING: Logging before InitGoogleLogging() is written to STDERR
W1013 20:00:39.671665  4650 _caffe.cpp:122] DEPRECATION WARNING - deprecated use of Python interface
W1013 20:00:39.671728  4650 _caffe.cpp:123] Use this instead (with the named "weights" parameter):
W1013 20:00:39.671738  4650 _caffe.cpp:125] Net('models/ucf101/tsn_bn_inception_rgb_deploy.prototxt', 1, weights='models/ucf101_split_1_tsn_rgb_reference_bn_inception.caffemodel')
[libprotobuf ERROR google/protobuf/text_format.cc:245] Error parsing text-format caffe.NetParameter: 19:12: Message type "caffe.LayerParameter" has no field named "bn_param".
F1013 20:00:39.673532  4650 upgrade_proto.cpp:88] Check failed: ReadProtoFromTextFile(param_file, param) Failed to parse NetParameter file: models/ucf101/tsn_bn_inception_rgb_deploy.prototxt
*** Check failure stack trace: ***
Aborted (core dumped)

**Comment by Aravinda89** [comment]
caffe-action is already built in lib/caffe-action.. Still getting this error

frame folder analysis done
WARNING: Logging before InitGoogleLogging() is written to STDERR
W1013 20:20:22.125752 16455 _caffe.cpp:122] DEPRECATION WARNING - deprecated use of Python interface
W1013 20:20:22.125816 16455 _caffe.cpp:123] Use this instead (with the named "weights" parameter):
W1013 20:20:22.125823 16455 _caffe.cpp:125] Net('models/ucf101/tsn_bn_inception_rgb_deploy.prototxt', 1, weights='models/ucf101_split_1_tsn_rgb_reference_bn_inception.caffemodel')
[libprotobuf ERROR google/protobuf/text_format.cc:245] Error parsing text-format caffe.NetParameter: 19:12: Message type "caffe.LayerParameter" has no field named "bn_param".
F1013 20:20:22.127086 16455 upgrade_proto.cpp:88] Check failed: ReadProtoFromTextFile(param_file, param) Failed to parse NetParameter file: models/ucf101/tsn_bn_inception_rgb_deploy.prototxt
*** Check failure stack trace: ***
Aborted (core dumped)

**Comment by yjxiong** [comment]
@Aravinda89 
You are not using the Caffe you built under caffe-action. See this message
```
DEPRECATION WARNING - deprecated use of Python interface
```

You may need to check your environment settings so that other installations do intervene.

**Comment by mlizhardy** [comment]
@Aravinda89  how do you check and set your environment settings ?
